pkg/client/outofband/client.go

```diff
@@ -67,8 +63,8 @@ func New(p Provider) (*Client, error) {
 // At least one attachment must be provided.
 // Service entries can be optionally provided. If none are provided then a new one will be automatically created for
 // you.
-func (c *Client) CreateRequest(opts ...RequestOptions) (*Request, error) {
-	req := &Request{}
+func (c *Client) CreateRequest(opts ...RequestOptions) (*outofband.Request, error) {
```

Comment:
We shouldn't ideally expose service pkg to the clients. We can wrap these structs in client pkg (similar to [didex](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/client/didexchange/models.go#L51)).

Reply:
@rolsonquadras  How does it help? It makes work even complicated.
For example, the usage below shows that you need to import two packages anyway.
```
didexchange1.Invitation{
    Invitation: &didexchange2.Invitation{ID:"1234"},
}
```

Sure, we can do something like that
```
inv := didexchange1.Invitation{}
inv.ID = "1234"
```

If we do not want to expose service pkg than we need to create package e.g `model` which can keep both client and service `types/models`. What we did in did-exchange can work only when we do read operation. But if we want to populate the structure we need to import the service package.

Reply:
@rolsonquadras I don't understand the benefits this actually brings in practice. Can you elaborate?

Reply:
Whole idea was to have the consumer use/look in client/framework pkg only (may be some util from framework). There was a discussion on this in one of the issues/PR. I'll try to find that.

Reply:
Here it is - https://github.com/hyperledger/aries-framework-go/pull/525#discussion_r337071685.

Reply:
There isn't any dicussion there.

Nevertheless I refactored - please take a look.

We're wasting too much time on nitpicks.